### DISCUSSION TITLE
BorgBackupPath

### DISCUSSION THREAD

**Discussion by GUNT0815** [nextcloud/all-in-one#2371] [Questions] [answered]
I run Nextcloud-AIO on Unraid. All works fine. But when I tried to update from v4.7.0 to the next version borg-backup failed.

Borgbackup log:
```
-------------------------------------------------------------
Performing backup...
Starting the backup...
Creating archive at "/mnt/borgbackup/borg::20230417_113103-nextcloud-aio"
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/borg/repository.py", line 1453, in get_fd
    ts, fd = self.fds[segment]
  File "/usr/lib/python3.10/site-packages/borg/lrucache.py", line 21, in __getitem__
    value = self._cache[key]  # raise KeyError if not found
KeyError: 291

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 5159, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 5090, in run
    return set_ec(func(args))
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 183, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 675, in do_create
    create_inner(archive, cache, fso)
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 642, in create_inner
    str(cache),
  File "/usr/lib/python3.10/site-packages/borg/cache.py", line 418, in __str__
    return self.str_format.format(self.format_tuple())
  File "/usr/lib/python3.10/site-packages/borg/cache.py", line 445, in format_tuple
    stats = self.stats()
  File "/usr/lib/python3.10/site-packages/borg/cache.py", line 435, in stats
    archive = Archive(self.repository, self.key, self.manifest, archive_name,
  File "/usr/lib/python3.10/site-packages/borg/archive.py", line 486, in __init__
    self.load(info.id)
  File "/usr/lib/python3.10/site-packages/borg/archive.py", line 497, in load
    self.metadata = self._load_meta(self.id)
  File "/usr/lib/python3.10/site-packages/borg/archive.py", line 489, in _load_meta
    data = self.key.decrypt(id, self.repository.get(id))
  File "/usr/lib/python3.10/site-packages/borg/repository.py", line 1203, in get
    return self.io.read(segment, offset, id)
  File "/usr/lib/python3.10/site-packages/borg/repository.py", line 1569, in read
    fd = self.get_fd(segment)
  File "/usr/lib/python3.10/site-packages/borg/repository.py", line 1455, in get_fd
    fd = open_fd()
  File "/usr/lib/python3.10/site-packages/borg/repository.py", line 1436, in open_fd
    fd = open(self.segment_filename(segment), 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/mnt/borgbackup/borg/data/0/291'
...
-------------------------------------------------------------
```

So I checked what path is configured for borg to backup to in the configuration.json:

-------------------------------------------------------------
"borg_backup_host_location": "/mnt/user/archive/nextcloud-borgbackup",
-------------------------------------------------------------

And also the nextcloud-aio webui shows:

--------------------------------------------
The backup itself will use a tool that is called [BorgBackup](https://github.com/borgbackup/borg#what-is-borgbackup) which is a well-known server backup tool that efficiently backs up your files and encrypts them on the fly.

Backups get created in the following directory on the host: /mnt/user/archive/nextcloud-borgbackup/borg
---------------------------------------------

So I checked in the console if the path really exists - yes, it is there and it and its subfolrder belong to root.

How can I get the borgbackup back to work? Do I have to change the owner of the folder?

**Comment by szaimen** [comment]
Hi, can you do an inregrity check of the backup?

**Comment by GUNT0815** [comment]
Integrity is checked with many errors (see below). What can I do to just start with the backups from scratch? Just delete all backup files?

----------------------------
Checking the backup integrity...
Starting repository check
finished segment check at segment 358
Starting repository index check
Index object count mismatch.
committed index: 323602 objects
rebuilt index:   302139 objects
ID: 0d09c95f655e0c2ddc6060c34af041033784955a1a9ef8688b362a9c838ebb6e rebuilt index: (268, 32136796)  committed index: <not found>     
ID: 3582c8d077cedbc4a017f0a73c3b5d51dc4059e53d8e54eaee4d5d8417d45593 rebuilt index: (268, 310239190) committed index: <not found>     
ID: 82dd5d6b76dc86d43fc6a83ae4e98999f74e9b9ab49ab59e91332f6d5e0ffcd9 rebuilt index: (268, 405709761) committed index: <not found>     
ID: b4e4a2c6c40ea67beb3630c55c57d2beec227d2a3966ae651882eb64c85fdb11 rebuilt index: (268, 467012551) committed index: <not found>     
ID: 71fb4396343d5099d412275153456dd12ba3469e6fe2ff92763bbe7b7db1f0fc rebuilt index: (268, 272516005) committed index: <not found>   
[...]
ID: 66043ead6b9c633f36869c4a7baa9e99485a50dc83068f6a801189a03e7aadef rebuilt index: <not found>      committed index: (305, 296879723)
ID: efffd6f88d71a4320fae5eb11f5d4cef66a9788d10d28ea8211b6b0e2f567a34 rebuilt index: <not found>      committed index: (302, 2648469)  
Finished full repository check, errors found.
Some errors were found while checking the backup integrity!

**Reply by szaimen** [reply]
> Integrity is checked with many errors (see below). What can I do

There should be now some advices shown in the AIO interface what you can do now.

**Comment by GUNT0815** [comment, marked as answer]
- Checked the integrity
- started correction -> successful
- started new backup attempt -> successful
- Rodger - discussion can be closed

Thank you for your help - thought I had read enough in the documentation - but there are still some surprises/good tools in the nextcloud-aio-gui ;-) Grüsse nach Berlin!

**Reply by szaimen** [reply]
Glad that you were able to resolve this! Grüße zurück! :)
